### PR TITLE
Issue #73 feat(cli): --init command creates a yaml config file

### PR DESCRIPTION
**Pull request by ryzzaki**
Running the `--init` command will now create `.dxscannerrc.yaml` file IF it does not exist yet (this is a condition to prevent file overwriting). The file is created in the relevant project folder and pre-loaded with all available practices listed in DXS (all practices are commented out with `#`) with their relevant metadata - this logic is not hardcoded, therefore, practice addition process for DX developers remains the same as the function pulls the metadata from the root container that is initially created.

### CODE REVIEW

**Comment by ryzzaki** [comment]
@prokopsimek please review the changes and let me know whether this is an acceptable solution within the architecture of the project and if it correctly fits your specifications described in issue #73.

File: src/init.configurator.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const yamlExists: boolean = fs.existsSync(yamlPath);
+  if (!yamlExists) {
+    createYAML(yamlPath, container);
+  }
+  cli.action.stop();
+};
+
+const createYAML = (yamlPath: string, container: Container) => {
+  let yamlInitContent = `# practices:`;
+  const initializedPractices: ContainerPracticeMetadata[] = container.getAll(Types.Practice);
```
Comment (prokopsimek) [inline src/init.configurator.ts]: I guess  you can use `IPracticeWithMetadata` interface.

```suggestion
  const initializedPractices = container.getAll<IPracticeWithMetadata[]>(Types.Practice);
```

File: src/init.configurator.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    yamlInitContent += `\n#    ${dataObject.id}: ${dataObject.impact}`;
+  }
+  fs.writeFile(yamlPath, yamlInitContent, (err) => {
+    if (err) {
+      throw ErrorFactory.newInternalError(err.message);
+    }
+  });
+};
+
+export interface ContainerPracticeMetadata {
```
Comment (prokopsimek) [inline src/init.configurator.ts]: you can use `IPracticeWithMetadata`

File: src/init.configurator.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  }
+  cli.action.stop();
+};
+
+const createYAML = (yamlPath: string, container: Container) => {
+  let yamlInitContent = `# practices:`;
+  const initializedPractices: ContainerPracticeMetadata[] = container.getAll(Types.Practice);
+  for (const practice of initializedPractices) {
+    const dataObject = practice.getMetadata();
+    yamlInitContent += `\n#    ${dataObject.id}: ${dataObject.impact}`;
```
Comment (prokopsimek) [inline src/init.configurator.ts]: Please, sort all practices alphabetically.

**Comment by prokopsimek** [comment]
@ryzzaki It's exactly what the #73 says! Thanks :) I've added few comments.

File: src/init.configurator.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import * as fs from 'fs';
+import { Container } from 'inversify';
+import { Types } from './types';
+import { cli } from 'cli-ux';
+import { IPracticeWithMetadata } from './practices/DxPracticeDecorator';
+
+export const initialize = (scanPath: string, container: Container) => {
+  const yamlPath = `${scanPath}/.dxscannerrc.yaml`;
+  cli.action.start(`Initializing configuration: ${yamlPath}`);
+  // check if .dxscannerrc.yaml already exists
```
Comment (adelkahomolova) [inline src/init.configurator.ts]: Check also if there are .dxscannerrc.yml or .dxscannerrc.json. See ConfigProvider.ts

Comment (prokopsimek) [reply src/init.configurator.ts]: Good catch! 👍 And the `.dxscannerrc` without an extension (for future purposes).

Comment (adelkahomolova) [reply src/init.configurator.ts]: I was also thinking about the file without extension but we don't check it in ConfigProvider either. So maybe we should :)

Comment (adelkahomolova) [reply src/init.configurator.ts]: I was thinking about this also and you're right. I checked ConfigProvider and didn't notice one condition, so we check it also there. :)

**Comment by ryzzaki** [comment]
I have added the pre-conditioning statements to check for the files stated by @adelkahomolova and @prokopsimek and I have also added one more integration test after @prokopsimek fixed my horrible test suite. All the test suites are passing 58/58 and the branch is ready for a merge with master. I have re-requested a final review from @adelkahomolova

**Pull request merged**